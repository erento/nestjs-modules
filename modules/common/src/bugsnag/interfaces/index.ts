export enum BugsnagSeverity {
    INFO = 'info',
    WARNING = 'warn',
    ERROR = 'error',
}
