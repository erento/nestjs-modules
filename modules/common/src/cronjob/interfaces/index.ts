export interface CronJobMethodReference {
    target: Object;
    methodName: string;
}
