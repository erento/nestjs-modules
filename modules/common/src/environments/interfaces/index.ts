export enum EnvironmentType {
    PROD = 'production',
    BETA = 'beta', // will be treated as production
    DEV = 'development',
    TEST = 'test',
}
