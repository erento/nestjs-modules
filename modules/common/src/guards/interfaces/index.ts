export enum AuthorizationType {
    service = 'service',
    admin = 'admin',
    seller = 'seller',
}
