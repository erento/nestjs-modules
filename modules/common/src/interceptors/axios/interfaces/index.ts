export type AxiosFulfilledInterceptor<T> = (value: T) => T | Promise<T>;
export type AxiosRejectedInterceptor = (error: any) => any;
