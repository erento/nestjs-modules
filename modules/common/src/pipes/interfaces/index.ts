export interface PaginationParams {
    page: number;
    size: number;
}

export interface PaginatedResponse<T> {
    page: number;
    pages: number;
    size: number;
    total: number;
    results: T[];
}

export interface OffsetAndLimit {
    offset?: number;
    limit?: number;
}
