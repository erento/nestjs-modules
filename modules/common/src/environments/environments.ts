import {EnvironmentType} from './interfaces';

export class Environments {
    public static getEnv (): EnvironmentType {
        return process.env.NODE_ENV === EnvironmentType.PROD ?
            EnvironmentType.PROD :
            process.env.NODE_ENV === EnvironmentType.TEST ?
                EnvironmentType.TEST :
                process.env.NODE_ENV === EnvironmentType.BETA ?
                    EnvironmentType.BETA :
                    EnvironmentType.DEV;
    }

    public static isProd (): boolean {
        return this.getEnv() === EnvironmentType.PROD || this.getEnv() === EnvironmentType.BETA;
    }

    public static isDev (): boolean {
        return this.getEnv() === EnvironmentType.DEV;
    }

    public static isTest (): boolean {
        return this.getEnv() === EnvironmentType.TEST;
    }

    public static getPackageJson (): any {
        // tslint:disable-next-line no-var-requires no-require-imports
        return require(process.cwd() + '/package.json');
    }

    public static getReleaseStage (): EnvironmentType {
        return this.getEnv();
    }

    public static getVersion (): string {
        return process.env.APP_VERSION || EnvironmentType.DEV;
    }
}
