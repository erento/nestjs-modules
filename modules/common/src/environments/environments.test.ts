import {Environments} from './environments';
import {EnvironmentType} from './interfaces';

describe('Environments', (): void => {
    test('should determine environment correctly', (): void => {
        const oldEnv: string | undefined = process.env.NODE_ENV;

        process.env.NODE_ENV = 'undefined-value';
        expect(Environments.getEnv()).toBe(EnvironmentType.DEV);
        expect(Environments.isProd()).toBe(false);
        expect(Environments.isTest()).toBe(false);
        expect(Environments.isDev()).toBe(true);

        process.env.NODE_ENV = 'development';
        expect(Environments.getEnv()).toBe(EnvironmentType.DEV);
        expect(Environments.isProd()).toBe(false);
        expect(Environments.isTest()).toBe(false);
        expect(Environments.isDev()).toBe(true);

        process.env.NODE_ENV = 'production';
        expect(Environments.getEnv()).toBe(EnvironmentType.PROD);
        expect(Environments.isProd()).toBe(true);
        expect(Environments.isTest()).toBe(false);
        expect(Environments.isDev()).toBe(false);

        process.env.NODE_ENV = 'test';
        expect(Environments.getEnv()).toBe(EnvironmentType.TEST);
        expect(Environments.isProd()).toBe(false);
        expect(Environments.isTest()).toBe(true);
        expect(Environments.isDev()).toBe(false);

        process.env.NODE_ENV = 'beta';
        expect(Environments.getEnv()).toBe(EnvironmentType.BETA);
        expect(Environments.isProd()).toBe(true);
        expect(Environments.isTest()).toBe(false);
        expect(Environments.isDev()).toBe(false);

        process.env.NODE_ENV = oldEnv;
    });
});
