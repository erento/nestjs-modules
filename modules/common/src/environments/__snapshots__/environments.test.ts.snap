// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environments should get keys 1`] = `"f85a2d2409fa127df8ac0d51b4c81f0d"`;
